go/.pre-commit-config.yaml

```diff
@@ -0,0 +1,26 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+files: ^go/
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: golangci-lint
+        entry: bash -c 'cd iceberg && golangci-lint run --fix --timeout 5m'
```

Comment:
does this require any installation?

Reply:
As with pyiceberg, you need to `pip install pre-commit` first and then you can run `pre-commit install` to get what you need.

Reply:
I think for someone who isn't actively working on pyiceberg (like me) this isn't obvious what exactly is required for installation, so would be good to add this somewhere in the docs. This can be a follow-up and can be part of the Docs that describe how to set up IntelliJ with Go and such, so that non-Go committers can review code and know how to run tests

Reply:
That makes sense, I can add stuff to the README for it. Though I have no idea how to set up IntelliJ with Go since I use VSCode lol, so i welcome any contributions on that :smile:

Reply:
So the thing is that most reviewers will be using Intellij and having some docs on how to set up stuff for Go would be great and also improves review cycles